### PR TITLE
FEATURE: add "Approve new topics unless user level" setting

### PR DESCRIPTION
**Pull request by pmusaraj**

### CODE REVIEW

File: lib/new_post_manager.rb [hunk 1]
```diff
@@ -73,6 +73,7 @@ def self.user_needs_approval?(manager)
 
     (user.trust_level <= TrustLevel.levels[:basic] && user.post_count < SiteSetting.approve_post_count) ||
     (user.trust_level < SiteSetting.approve_unless_trust_level.to_i) ||
+    (manager.args[:title].present? && user.trust_level < SiteSetting.approve_new_topics_unless_trust_level.to_i) ||
```
Comment (ZogStriP) [inline lib/new_post_manager.rb]: This looks good but definitely needs a test.

Comment (pmusaraj) [inline lib/new_post_manager.rb]: Sure. Are you looking for tests in Ruby or javascript? (I'm not too familiar with tests for either, but definitely less familiar with Ruby testing.)

Comment (ZogStriP) [inline lib/new_post_manager.rb]: Ruby tests. You'll need to add a test case in [spec/components/new_post_manager_spec.rb](https://github.com/discourse/discourse/blob/master/spec/components/new_post_manager_spec.rb) ;)

Comment (pmusaraj) [inline lib/new_post_manager.rb]: Tests added (and they helped me find another needed fix). It's my first rails test, so please check carefully.

**Comment by SamSaffron** [comment]
Sure... lets pull this in, thanks @pmusaraj  !

**Pull request merged**